File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        // count the amount of times each number appears using a hashmap
        // decide between set and map -> map easier to expand to other problems set makes more sense for this problem
        // if you have already seen the number then return false
        // if you get all the way through then there are no duplicates

        HashSet<Integer> count = new HashSet<>();

        for(Integer num : nums){
            if(count.contains(num)){
                return true;
            } else {
                count.add(num);
            }
        }
        
        return false;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        longestSequence = 0

        for num in numSet:
            if num - 1 not in numSet:
                i = num
                sequence = 1
                while i + 1 in numSet:
                    i += 1
                    sequence += 1
                if sequence > longestSequence:
                    longestSequence = sequence

        return longestSequence